clj/src/com/sixsq/slipstream/clj_client/run.clj

```diff
@@ -1,238 +1,257 @@
-;   (get "http://aoeu.com/1")
-;   (post "http://aoeu.com/2")
-;   (get "http://aoeu.com/3")
-;   (get "http://aoeu.com/999"))
-
-;; Default set of request parameters.
-(def ^:private  rtp-req-params (conj
-                     ru/base-req-params
-                     {:content-type "text/plain"
-                      :accept "text/plain;charset=utf-8"
-                      :query-params {:ignoreabort "true"}}))
-(def ^:private scale-req-params (assoc rtp-req-params :accept "application/json"))
+  (:use [slingshot.slingshot :only [try+]]))
+
 
 ;;
-;; Public library
+;; Public library API.
 
-(defn get-rtp
-  "Get runtime paramter.
-  "
-  [node-name id param]
+
+;; Getters and setters of component, component instance and global parameters.
+(defn get-param
+  "Get parameter 'param' of application component instance 'comp.id' as 'comp.id:param'.
+  When 'id' is nil, gets parameter of the application component as 'comp:param'."
+  [comp id param]
   (:body
     (http/get
-      (ru/build-rtp-url node-name id param)
-      rtp-req-params)))
+      (ri/build-param-url comp id param)
+      ri/param-req-params)))
 
-(defn set-rtp
-  "Set runtime parameter.
-  "
-  [node-name id param value]
+(defn set-param
+  "Set parameter 'param' to 'value' on application component instance 'comp.id'."
+  [comp id param value]
   (http/put
-    (ru/build-rtp-url node-name id param)
-    (merge rtp-req-params {:body value})))
+    (ri/build-param-url comp id param)
+    (merge ri/param-req-params {:body value})))
+
+(defn get-scale-state
+  "Get scale state of the component instance."
+  [comp id]
+  (get-param comp id "scale.state"))
 
 (defn get-state
+  "Get state of the run."
   []
   (:body
     (http/get
-      ru/run-state-url
-      rtp-req-params)))
+      ri/run-state-url
+      ri/param-req-params)))
 
 (defn get-abort
+  "Get abort message."
   []
   (try+
-    (:body (http/get ru/run-abort-url rtp-req-params))
+    (:body (http/get ri/run-abort-url ri/param-req-params))
     (catch [:status 412] {} "")))
 
-(defn get-node-multiplicity
-  [node-name]
-  (Integer/parseInt (get-rtp node-name nil "multiplicity")))
+(defn get-multiplicity
+  "Get multiplicity of application component 'comp'."
+  [comp]
+  (Integer/parseInt (get-param comp nil "multiplicity")))
+
+(defn get-comp-ids
+  "Get list of instance IDs of application component 'comp'."
+  [comp]
+  (remove #(zero? (count %))
+          (-> (try+
+                (get-param comp nil "ids")
+                (catch [:status 412] {} ""))
+              (u/split #",")
+              (sort))))
 
-(defn get-node-ids
-  "IDs of the node as list."
-  [node-name]
-  (->> (-> (try+
-             (get-rtp node-name nil "ids")
-             (catch [:status 412] {}  ""))
-           (u/split  #",")
-           (sort))
-       (remove #(= (count %) 0))))
 
 ;; Predicates.
 (defn aborted?
+  "Check if run is in \"Aborted\" state."
   []
-  (not (empty? (get-abort))))
+  (not (s/blank? (get-abort))))
```

Comment:
`get-abort` gets from the server and returns the abort message. Empty - no abort, not empty - abort is set.  Server side, by definition, the abort is raised is when the abort message is not empty. IMHO, `(not (s/blank? (get-abort))))` is OK, until this issue is worked on https://github.com/slipstream/SlipStreamClientAPI/issues/9